File: examples/nodejs/src/network.ts
import { post } from './sock.js'

(async () => {
  const json = await post('/internal/network', {
    message: {
      "recipient_dids": ["123", "456"],
      "hub_endpoint": "https://hub.example.com",
      "heartbeat": 123,
      "trm": "true",
    }
  })

  console.log(json)
})()

File: agent/src/controllers/internal/network.rs
use crate::{controllers::errors::AgentErrorCode, services::studio::Studio};
use axum::extract::Json;
use serde::{Deserialize, Serialize};
use serde_json::Value;

// NOTE: POST /internal/version
#[derive(Deserialize, Serialize)]
pub struct MessageContainer {
    message: Value,
}

pub async fn handler(
    Json(_): Json<MessageContainer>,
) -> Result<Json<&'static str>, AgentErrorCode> {
    let mut studio = Studio::new();
    match studio.network().await {
        Ok(_) => Ok(Json("ok")),
        Err(e) => {
            log::error!("{:?}", e);
            Err(AgentErrorCode::NetworkInternal)?
        }
    }
}
